### PR TITLE
Integration-library: merges additions from PR #2657 on latest library

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

This PR catches up with some unmerged changes.

preamble:
 - PR #2657 was created by @GitHK (long time ago) and never merged to master
 - PR #2657 is already used in a released version of  https://github.com/ITISFoundation/ci-service-integration-library **NOTE**: we should avoid this to happen again! :-(
 - From commit [``c61dbd5``](https://github.com/ITISFoundation/osparc-simcore/tree/c61dbd5baf2ff865dbb74eb7e2ed0b031c0f3b9e) we have an upgraded ``integration-library`` (``ooil``)
 - we want the new version to include this functionality so it is still compatible with  https://github.com/ITISFoundation/ci-service-integration-library. 

therefore, this PR:
 - merges PR #2657 on top of ``c61dbd5``
 - fixes issues detected by tooling (linter, sonarcloud and codeclimate)
-  does NOT add any changes to the design. For the sake of clarity, those will be done in a separate PR.

## Related issue/s

- closes  #2657

### CODE REVIEW

**Review by GitHK — APPROVED** [review]
Thanks a lot for bringing it in. Looks like all is there.

File: packages/service-integration/src/service_integration/commands/config.py [hunk 1]
```diff
@@ -88,11 +88,19 @@ def main(
...
 
-    create_osparc_specs(from_spec_file, meta_cfg_path, runtime_cfg_path)
+    create_osparc_specs(
+        from_spec_file, project_cfg_path, meta_cfg_path, runtime_cfg_path
+    )
+
+
+if __name__ == "__main__":
+    # pylint: disable=no-value-for-parameter
+    main()
```
Comment (GitHK) [inline packages/service-integration/src/service_integration/commands/config.py:106]: I guess you want this here.

**Review by GitHK — APPROVED** [review]
👍

**Review by sanderegg — APPROVED** [review]
great job! thanks !

File: packages/service-integration/src/service_integration/commands/config.py (hunk 1 above)
Comment (pcrespov) [reply packages/service-integration/src/service_integration/commands/config.py:106]: i will remove those in the next PR ... it was to test locally

**Pull request merged**